packages/flutter_tools/test/general.shard/windows/visual_studio_validator_test.dart

```diff
@@ -19,19 +19,33 @@ void main() {
 
     setUp(() {
       mockVisualStudio = MockVisualStudio();
-      // Mock a valid VS installation.
+      // Default values regardless of whether VS is installed or not.
+      when(mockVisualStudio.workloadDescription).thenReturn('Desktop development');
+      when(mockVisualStudio.necessaryComponentDescriptions(any)).thenReturn(<String>['A', 'B']);
+    });
+
+    // Assigns default values for a complete VS installation with necessary components.
+    void _configureMockVisualStudioAsInstalled() {
       when(mockVisualStudio.isInstalled).thenReturn(true);
       when(mockVisualStudio.isPrerelease).thenReturn(false);
       when(mockVisualStudio.isComplete).thenReturn(true);
       when(mockVisualStudio.isLaunchable).thenReturn(true);
       when(mockVisualStudio.isRebootRequired).thenReturn(false);
       when(mockVisualStudio.hasNecessaryComponents).thenReturn(true);
-      when(mockVisualStudio.workloadDescription).thenReturn('Desktop development');
-      when(mockVisualStudio.necessaryComponentDescriptions(any)).thenReturn(<String>['A', 'B']);
-      when(mockVisualStudio.fullVersion).thenReturn('15.1');
```

Comment:
I think codecoverage is unhappy because you aren't ever setting this now, which means you are only ever testing the null codepath, which is not good (e.g., if a bug were introduced in the 'split' logic).

`_configureMockVisualStudioAsInstalled` should be setting a version so that it's behaving like an actual install would.

It wouldn't hurt to add a test of the 2017 vs 2019 message difference to ensure that both are being covered and catch mock configuration issues like this.

Reply:
SG, I added the version name and number to the "installed" test and check that it's included in the message. Also added the check for the right tool version given the major version, but in visual_studio_test.dart since it's all mocked in the validator tests.